text/0066-separate-context-propagation.md

```diff
@@ -0,0 +1,557 @@
+
+Note that OpenTelemetry APIs calls should *always* be given access to the entire 
+context object, and never just a subset of the context, such as the value in a 
+single key. This allows the SDK to make improvements and leverage additional 
+data that may be available, without changes to all of the call sites.
+
+## Correlations API
+
+In addition to trace propagation, OpenTelemetry provides a simple mechanism for 
+propagating metric labels, called the Correlations API. Correlations are 
+intended for indexing events in one service with attributes provided by a prior 
+service in the same transaction. This helps to establish a causal relationship 
+between these events. For example, determining that a particular browser version 
+is associated with a failure in an image processing service.
+
+Correlations are implemented as an extremely basic cross-cutting concern - a 
+transaction-level dictionary. All key-value pairs are stored in the context 
+object, and then injected by the Correlation propagator to be sent to the next 
+process. 
+
+The Correlations API is based on the [W3C Correlation-Context specification](https://w3c.github.io/correlation-context/), 
+and implements the protocol as it is defined in that working group.
+
+While Correlations can be used to prototype other cross-cutting concerns, this 
+mechanism is primarily intended to convey values for the OpenTelemetry 
+observability systems. New concerns with different criteria should be modeled 
+separately, using the same underlying context propagation layer as building 
+blocks.
+
+For backwards compatibility, OpenTracing Baggage is propagated as Correlations 
+when using the OpenTracing bridge.
+
+As every correlation adds additional overhead to every request, correlations 
+should be used with care. The implementation MUST propagate these values. It is 
+entirely on the user to choose when to drop correlations, either by removing 
+them via the API, or by filtering them via a network proxy or service mesh.
+
+**`GetCorrelation(context, key) -> value`**  
+To access the value for a label set by a prior event, the Correlations API 
+provides a function which takes a context and a key as input, and returns a value.
+
+**`SetCorrelation(context, key, value) -> context`**  
+To record the value for a label, the Correlations API provides a function which 
+takes a context, a key, and a value as input, and returns an updated context 
+which contains the new value.
+
+**`RemoveCorrelation(context, key) -> context`**  
+To delete a label, the Correlations API provides a function 
+which takes a context and a key as input, and returns an updated context which 
+no longer contains the selected key-value pair.
+
+**`ClearCorrelations(context) -> context`**  
+To avoid sending any labels to an untrusted process, the Correlation API 
+provides a function to remove all Correlations from a context.
+
+**`GetCorrelationPropagator() -> (HTTP_Extractor, HTTP_Injector)`**  
```

Comment:
why are they called HTTP_Extractor? Please mention binary extractors as well

Reply:
I left Binary extractors out the OTEP for conciseness, as they are basically the same as HTTP propagators. But I do want to make it clear that there is no universal propagator, only propagators for specific protocols. I can add them back if you like.